scanpipe/pipelines/deploy_to_develop.py

```diff
@@ -197,6 +203,23 @@ def flag_mapped_resources_archives_and_ignored_directories(self):
         flag.flag_ignored_directories(self.project)
         d2d.flag_processed_archives(self.project)
 
+    def perform_janitorial_tasks(self):
+        """
+        On deployed side
+            - PurlDB match files with ``no-java-source``, ``too-many-maps`` and empty
+                status, if no match is found update status to ``requires-review``.
+            - Update status for uninteresting files.
+
+        On devel side
+            - Update status for not deployed files.
+        """
+        d2d.perform_janitorial_tasks(
+            project=self.project,
+            matched_extensions=self.purldb_resource_extensions,
+            uninteresting_extensions=self.uninteresting_extensions,
+            logger=self.log,
+        )
```

Comment:
What about 3 specialized and reusable functions instead of a single one that does a lot?